### PR TITLE
Extrafields

### PR DESCRIPTION
**Pull request by lentregu**
Adding optional flags for writing filename and number and function

### CODE REVIEW

File: logger.go [hunk 1]
```diff
@@ -51,10 +57,22 @@ func (l *Logger) SetWriter(w io.Writer) {
 	l.mu.Unlock()
 }
 
+func (l *Logger) SetFlags(flags int32) {
+	l.mu.Lock()
+	l.flags = l.flags | flags
```
Comment (crbrox) [inline logger.go]: not sure if would be better  to set all the  flags at once (a simple assignment) ... maybe adding a AddFlags method for doing the OR ...  What do you think?

File: logger.go [hunk 2]
```diff
@@ -51,10 +57,22 @@ func (l *Logger) SetWriter(w io.Writer) {
...
+	l.mu.Unlock()
+}
+
 func (l *Logger) format(buffer *bytes.Buffer, lline logLine) {
 	var dynamicContext C
 	now := time.Now()
-	fmt.Fprintf(buffer, prefixFormat, now.Format(timeFormat), levelNames[lline.level])
+
+	var flagsFields string
+	if l.flags&(Llongfile|Lshortfile|Lmethod) != 0 {
```
Comment (crbrox) [inline logger.go]: I think l.flags shouldn't be read without acquiring the mutex first  ... An even better alternative could be use an atomic int32 as we do with `level`

**Comment by crbrox** [comment]
Terrific work!

File: logger.go (hunk 1 above)
Comment (lentregu) [inline logger.go]: The SetFlags is changed to set a flag and another AddFlags method has been added to allow adding flags without taking off the existing ones.

File: logger.go (hunk 2 above)
Comment (lentregu) [inline logger.go]: flags access now is taking out by means of atomic.LoadInt32 and atomic.StoreInt32 methods to read and write respectively.

**Comment by crbrox** [comment]
LGTM

**Pull request merged**